LearningJourney/Modules/Features/JLibrary/JLibraryTests/Sources/Data/LibraryParserTests.swift

```diff
@@ -0,0 +1,50 @@
+import XCTest
+@testable import JLibrary
+
+final class LibraryParserTests: XCTestCase {
+    // MARK: - System under test
+
+    private let sut = LibraryParser()
+
+    // MARK: - Tests
+
+    func testParseDecodableDataShouldSucceed() throws {
+        // Given
+        let decodableData = DecodableData()
+        let encodedData = try JSONEncoder().encode(decodableData)
+
+        // When
+        let parseResult: Result<DecodableData, ParsingError> = sut.parse(encodedData)
+
+        // Then
+        switch parseResult {
+            case .success(let data):
+                XCTAssertEqual(data, decodableData)
+            case .failure(let error):
+                XCTFail("Expected decoded data but got \(error)")
+        }
+    }
+
+    func testParseUndecodableDataShouldFail() throws {
+        // Given
+        let undecodableData = UndecodableData()
+        let encodedData = try JSONEncoder().encode(undecodableData)
+
+        // When
+        let parseResult: Result<DecodableData, ParsingError> = sut.parse(encodedData)
+
+        // Then
+        switch parseResult {
+            case .success(let data):
+                XCTFail("Input data shouldn't be decodable but got \(data)")
+            case .failure:
+                break
+        }
+    }
+}
+
+private struct DecodableData: Encodable, Decodable, Equatable {
+    var areWeLucky: Bool = .random()
```

Comment:
You should **never** have random values in unit tests. Unit tests should be reproducible, a fixed known input should produce a fixed known output. This might not affect the context of this test, but we can't be sure that this is the only place that it'll be used (private modifier is a hint but not law)

If you want to indicate that something won't interfer with the test, you can just name it with a constant value
```
    var dummy : Bool { true }
```